packages/compartment-mapper/src/link.js

```diff
@@ -372,16 +411,42 @@ export const link = (
     // The `moduleMapHook` writes back to the compartment map.
     compartmentDescriptor.modules = modules;
 
+    /** @type {Record<string, Language>} */
+    const finalLanguageForExtension = Object.create(null);
+    for (const [extension, language] of entries(customLanguageForExtension)) {
+      if (extension in languageForExtension) {
+        throw new Error(
+          `Extension ${q(extension)} already assigned language ${q(languageForExtension[extension])}`,
+        );
+      }
+      finalLanguageForExtension[extension] = language;
+    }
+    for (const [extension, language] of entries(languageForExtension)) {
+      finalLanguageForExtension[extension] = language;
```

Comment:
I'm not sure if I should write anything back to the `CompartmentDescriptor`.

Reply:
No, this is fine.